### DISCUSSION TITLE
Expose Lima Machine Ports Lan

### DISCUSSION THREAD

**Discussion by wviana** [lima-vm/lima#813] [Q&A]
Hello,

I'm trying to setup k3s on my mac. I could setup it and deploy a hello word application. From inside de mac if I run `curl http://localhost:8080` it returns me the expected content.

By another computer in my lan, if I try `curl http://192.168.1.180:8080` ( mac ip )  I don't get any response. Is there a way to expose lima ports to lan.

**Comment by imhansai** [comment]
I'm configured like this. I don't know if it's a best practice, but it works normally

![image](https://user-images.githubusercontent.com/17228526/165210471-f926b766-15fc-4cd8-ae85-0bf19165b7bc.png)

**Comment by jandubois** [comment]
You can add a portforwarding rule to your `lima.yaml` file:

```yaml
portForwards:
- guestPortRange: [1, 65535]
  guestIPMustBeZero: true
  hostIP: 0.0.0.0
```

This will bind all ports that are bound to `0.0.0.0` inside the VM to `0.0.0.0` on the host (but all others still just to `127.0.0.1`).

If you leave out `guestIPMustBeZero: true`, then **all** forwarded ports will be bound to the external interface, which can be a security issue.

**Reply by wviana** [reply]
Sorry for not updating before. After @imhansai comment I've made these changes. Even like it I could not get from network. I think Mac may be asking for allowing application do expose the ports. But I'm accessing it by ssh, got lazy to open and see screen. Will chek other time and update you all.

**Reply by imhansai** [reply]
> -p 0.0.0.0:hostPort:guestPort is accessible from the outside interface,e.g. 192.168.1.111:hostPort
-p 127.0.0.1:hostPort:guestPort or -p hostPort:guestPort only accessible locally

Is this correct?

**Reply by jandubois** [reply]
> > -p 0.0.0.0:hostPort:guestPort is accessible from the outside interface,e.g. 192.168.1.111:hostPort
> > -p 127.0.0.1:hostPort:guestPort or -p hostPort:guestPort only accessible locally
> 
> Is this correct?

Yes, but `0.0.0.0` is the default, so that case can just be `-p hostport:guestport`

**Comment by wviana** [comment]
Yeah, that was it. Just clicked in and accept and now its working from LAN.
Probably will try something more restrictive when going for real, but for learning it's enoght.

Now I have to discover how to change kubeconfig.yaml to be able to run `kubectl` commands from LAN. Getting a `Unable to connect to the server: x509: certificate is valid for 10.43.0.1, 127.0.0.1, 192.168.5.15, not 192.168.1.118`.

Thank you all for helping. :smiley:
